core/roboconf-agent/src/main/java/net/roboconf/agent/internal/misc/UserDataUtils.java

```diff
@@ -166,6 +169,28 @@ public static AgentProperties findParametersForAzure( Logger logger ) {
 	}
 
 
+	/**
+	 * Reconfigures the messaging.
+	 * @param etcDir the KARAF_ETC directory
+	 * @param msgData the messaging configuration parameters
+	 */
+	public static void reconfigureMessaging(String etcDir, Map<String,String> msgData, String messagingType) throws IOException {
+		if( ! Utils.isEmptyOrWhitespaces(etcDir)) {
+			String fileName = "net.roboconf.messaging." + messagingType + ".cfg";
+			PrintWriter out = null;
+			try {
+				// Open file in overwrite mode (append=false)
+				out = new PrintWriter(new FileOutputStream(etcDir + File.separator + fileName), false);
+				for(String key : msgData.keySet()) {
```

Comment:
Using **Map.entrySet()** would be more efficient.
You should also verify Karaf's ETC directory exist.
And add a unit test too. :+1: